f600233d1b154d507b782f84604bcb12/update/sys_script_include_35c37d36977611106fa8b4b3f153af94.xml

```diff
@@ -0,0 +1,126 @@
+<?xml version="1.0" encoding="UTF-8"?><record_update table="sys_script_include">
+    <sys_script_include action="INSERT_OR_UPDATE">
+        <access>package_private</access>
+        <active>true</active>
+        <api_name>x_g_dis_atat.MultipartHelper</api_name>
+        <caller_access/>
+        <client_callable>false</client_callable>
+        <description>A class that helps with generating multipart requests for EDMS document uploads.</description>
+        <name>MultipartHelper</name>
+        <script><![CDATA[var MultipartHelper = Class.create();
+MultipartHelper.prototype = {
+  initialize: function() {
+  },
+  createBody: function(packageSysId, documentSysId, indexFields) {
+    var endpoint = gs.getProperty("glide.servlet.uri") + "api/x_g_dis_atat/edms/edms-multipart-attachment";
+    var indexFieldsString = JSON.stringify(indexFields);
+
+    var internalRequest = new sn_ws.RESTMessageV2();
+    internalRequest.setEndpoint(endpoint);
+    internalRequest.setHttpMethod("GET");
+    internalRequest.setRequestHeader("Accept", "multipart/form-data");
+    internalRequest.setBasicAuth(
+      gs.getProperty("x_g_dis_atat.edms.atatClient.username"),
+      gs.getProperty("x_g_dis_atat.edms.atatClient.password")
+    );
+
+    var boundary = "----" + gs.generateGUID() + packageSysId;
+
+    // query parameters
+    internalRequest.setQueryParameter("boundary", boundary);
+    internalRequest.setQueryParameter("packageId", packageSysId);
+    internalRequest.setQueryParameter("documentSysId", documentSysId);
+    internalRequest.setQueryParameter("indexFields", indexFieldsString);
+
+    internalRequest.saveResponseBodyAsAttachment(
+      'x_g_dis_atat_acquisition_package',
+      packageSysId,
+      'TemporaryMultipartContentFile.txt'
+    );
+
+    var response = internalRequest.execute();
```

Comment:
I do not believe we can just call it as a regular library because of certain methods (e.g., `.writeStreamWriter()`) that construct the request body are only available with REST request objects. Basically making an HTTP request provides access to the objects needed to construct the request body. At least that is my understanding and the API documentation for [RESTAPIResponse](https://developer.servicenow.com/dev.do#!/reference/api/rome/server/sn_ws-namespace/c_RESTAPIResponse) mentions it. 

> Note: You cannot instantiate objects of this type. Objects of this type are created automatically and are accessible only in scripted REST API resource scripts.

I'm not sure if there is a better way or alternative, and did not explore any.

Reply:
Thanks for the explainer and doc reference.

Reply:
OK I think I understand the crazy machinations going on here...or at least I don't want to rock the boat as ridiculous as it is to have to jump through these hoops.